lib/ui/page/home/page/chat/widget/data_attachment.dart

```diff
@@ -83,12 +83,13 @@ class _DataAttachmentState extends State<DataAttachment> {
         };
       } else if (e is LocalAttachment) {
         leading = switch (e.status.value) {
-          SendingStatus.sending => _Progress(
-            key: const Key('Sending'),
-            progress: e.progress.value,
-
-            // TODO: Remove when implement upload cancellation.
-            showCancelIcon: false,
+          SendingStatus.sending => InkWell(
+            key: const Key('CancelUploading'),
+            onTap: () => e.cancelUpload(),
```

Comment:
Why not just?


```suggestion
            onTap: e.cancelUpload,
```

Your solution creates an anonymous function, which still takes few CPU cycles in order to be executed as a subroutine. Instead a pointer to the existing function can be provided, can't it?